src/Client.js

```diff
@@ -0,0 +1,20 @@
+const httpClient = require('./http-client');
+const applicationDiscoveryService = require('./api/application-discovery-service');
+
+class Client {
+  constructor({ baseURL } = {}) {
+    if (!baseURL) {
+      throw new Error('baseURL is required');
+    }
+
+    this.httpClient = httpClient({ baseURL });
+
+    applicationDiscoveryService(this);
```

Comment:
So this Client will be a big one that holds every single OpenPaaS API possible like this, right?

```js
    applicationDiscoveryService(this);
    calendarService(this);
    inboxService(this);
```

In my opinion, that might not what we want for a frontend client, since using a single endpoint requires importing everything.

Reply:
How about having an abstract class `Client` and many `ADSClient`, `InboxClient`, `CalendarClient`, etc. that extends it. And then we'll only import the client we want to use.

Reply:
AFAIK, this client is responsible only for core APIs. For particular APIs like calendar, inbox, etc, the frontend client will be implemented in the corresponding SPAs.

Reply:
And that Client will have the methods to communicate with the equivalent endpoints.

Reply:
For more discussion detail: https://chat.linagora.com/openpaas/pl/633drbncb7y4mgqdnfw6err7fo

Reply:
I was just bringing some examples. Even the core APIs will be divided into multiple separate groups, so it would eventually become:

```
coreService1(this);
coreService2(this);
coreService3(this);
...
```

Reply:
And what if I just want to use ADS, but not the other core APIs, e.g. in the Application Grid web component.

Reply:
Agree with you

Reply:
I modified the code to able to load necessary APIs instead of all